test/opengl/testing.h

```diff
@@ -0,0 +1,93 @@
+#ifndef _TESTING_H_
+#define _TESTING_H_
+
+#include "Halide.h"
+#include <iostream>
+#include <exception>
+#include <functional>
+#include <cmath>
+
+namespace Testing {
+
+    // fill 3-dimension buffer
+    template <typename T>
+        void fill(Halide::Buffer<T> &buf, std::function<T(int x, int y, int c)> f)
```

Comment:
Indentation and bracing doesn't follow Halide conventions: This should really be

namespace Testing {

template<typename T>
void fill (...) {
   ...
}